python/lsst/obs/metadata/observationInfo.py

```diff
@@ -0,0 +1,190 @@
+                   "detector_name": ("Name of the detector within the instrument (might not be unique)",
+                                     "str"),
+                   "detector_exposure_id": ("Unique integer identifier for this detector in this exposure",
+                                            "int"),
+                   "object": ("Object of interest or field name", "str"),
+                   "temperature": ("Temperature outside the dome", "astropy.units.Quantity"),
+                   "pressure": ("Atmospheric pressure outside the dome", "astropy.units.Quantity"),
+                   "relative_humidity": ("Relative humidity outside the dome", "float"),
+                   "tracking_radec": ("Requested RA/Dec to track", "astropy.coordinates.SkyCoord"),
+                   "altaz_begin": ("Telescope boresight azimuth and elevation at start of observation",
+                                   "astropy.coordinates.AltAz"),
+                   "science_program": ("Observing program (survey or proposal) identifier", "str"),
+                   "obstype": ("Type of observation (currently: science, dark, flat, bias, focus)",
+                               "str"),
+                   "obsid": ("Label uniquely identifying this observation (can be related to 'exposure')",
+                             "str")}
+
+    """All the properties supported by this class with associated
+    documentation."""
+
+    def __init__(self, header, translator_class=None):
+
+        # Store the supplied header for later stripping
+        self._header = header
+
+        # PropertyList is not dict-like so force to a dict here to simplify
+        # the translation code.
+        if hasattr(header, "toOrderedDict"):
+            header = header.toOrderedDict()
+
+        if translator_class is None:
+            translator_class = MetadataTranslator.determineTranslator(header)
+        elif not issubclass(translator_class, MetadataTranslator):
+            raise TypeError(f"Translator class must be a MetadataTranslator, not {translator_class}")
+
+        # Create an instance for this header
+        translator = translator_class(header)
+
+        # Store the translator
+        self._translator = translator
+
+        # Loop over each translation (not final form -- this should be
+        # defined in one place and consistent with translation classes)
+
+        for t in self._PROPERTIES:
+            # prototype code
+            method = f"to_{t}"
+            property = f"_{t}"
+
+            try:
+                setattr(self, property, getattr(translator, method)())
+            except AttributeError:
+                log.warning("No translation exists for property %s using translator %s",
+                            t, translator.__class__)
+            except KeyError:
+                log.warning("Error calculating property %s using translator %s",
+                            t, translator.__class__, exc_info=1)
```

Comment:
I think this should be a `log.error` and a `raise`: if a value can't be translated, that feels like a serious bug.

Reply:
Also, looking at the coverage report says that this KeyError is never triggered. That's worth an explicit test in the TrivialTranslator test case.

Reply:
I'm a bit worried about raising here. I have gone either way on this as I've developed. A warning is roughly what happens with VisitInfo population at the moment if something is missing.  Raising is the right answer since it tells me that the translator needs fixing for this new type of weird header that has turned up. On the other hand people get cranky when they process 10,000 files and one of them falls over after and brings everything down five minutes after they leave work.

Reply:
If 1/10,000 files is messed up, we should probably know about it immediately. A warning means we won't almost certainly won't know until something further down the pipeline falls over because a value is incorrect.